### PR TITLE
Validate if username or email exists

### LINKED ISSUE #28: Add signup page
With referral functionality

### PR DESCRIPTION
**Pull request by JamesHageman**
Closes #28

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -53,26 +53,26 @@ export type BroadcastOwner = {
 
 export type BroadcastPage = 'explore' | 'home' | 'profile';
 
-export type Action =
```
Comment (JamesHageman) [inline src/actions.js]: I decided to reformat these actions to clean up the diffs when we add new ones.

File: src/components/header/Header.js [hunk 2]
```diff
@@ -10,34 +10,62 @@ class Header extends React.Component {
   props: Props;
   render() {
     return <header>
-    <div className="pure-menu pure-menu-horizontal header-wrapper">
```
Comment (JamesHageman) [inline src/components/header/Header.js]: I also reformatted the jsx in this file

File: src/sagas/index.js [hunk 3]
```diff
@@ -41,6 +41,52 @@ function* watchSignup() {
...
+  if (!errors.username && !errors.email) {
+    yield call(validateSignupFields, username, email);
+  };
+}
+
+function* validateSignupFields(username: string, email: string) {
+  try {
+    const res = yield call(api.userExists, username, email);
+    const { username_exists = false, email_exists = false } = res;
+    let errors = {};
```
Comment (ethan-hardy) [inline src/sagas/index.js]: You can make `errors` `const` here, since you're only adding keys to it

Comment (JamesHageman) [inline src/sagas/index.js]: tru

Comment (JamesHageman) [inline src/sagas/index.js]: fixed

**Comment by JamesHageman** [comment]
@ethan-hardy pls re-review

**Comment by ethan-hardy** [comment]
🎉 🎉

**Comment by georgeutsin** [comment]
screen shot 2016-11-24 at 9 39 48 pm
hovering over a broadcast just created errors out and doesn't even load the user afterwards

**Pull request merged**